### PR TITLE
Add E2E test for reduced ASM mode

### PR DESCRIPTION
**Pull request by swetharepakula**
* when ASM enabled, neg controller should create negs for services and will not trigger based on destination rules or subset

### CODE REVIEW

File: cmd/e2e-test/asm_test.go [hunk 1]
```diff
@@ -285,3 +287,95 @@ func TestNoIstioASM(t *testing.T) {
 
 	})
 }
+
+// TestGSMService tests the service/destinationrule, it can't run parallel with other tests.
+func TestGSMService(t *testing.T) {
+	_ = apiv1.ComponentStatus{}
```
Comment (bowei) [inline cmd/e2e-test/asm_test.go:293]: What is this used for?

File: cmd/e2e-test/asm_test.go [hunk 2]
```diff
@@ -285,3 +287,95 @@ func TestNoIstioASM(t *testing.T) {
...
+
+	// negStatus is used to verify GC in the zones NEGs were created. The actual negStatus is not important
+	// and therefore we can use the last generated neg status.
+	var negStatus *annotations.NegStatus
+
+	// This test case will need two namespaces, one will in asm-skip-namespaces.
+	Framework.RunWithSandbox("TestASMService", t, func(t *testing.T, sSkip *e2e.Sandbox) {
```
Comment (bowei) [inline cmd/e2e-test/asm_test.go:300]: I recommend extending the RunWithSandbox to be able to give multiple namespaces than nesting like this, which can make future maintenance tricky as I'm not sure what the semantics would be.

File: cmd/e2e-test/asm_test.go [hunk 3]
```diff
@@ -285,3 +287,95 @@ func TestNoIstioASM(t *testing.T) {
...
+		Framework.RunWithSandbox("TestASMService", t, func(t *testing.T, s *e2e.Sandbox) {
+			// Enable ASM mode
+			ctx := context.Background()
```
Comment (bowei) [inline cmd/e2e-test/asm_test.go:303]: Move this down to where it is used.

File: cmd/e2e-test/asm_test.go [hunk 4]
```diff
@@ -285,3 +287,95 @@ func TestNoIstioASM(t *testing.T) {
...
+
+			asmConfig := map[string]string{"enable-asm": "true",
+				"asm-skip-namespaces": fmt.Sprintf("kube-system,istio-system,%s", sSkip.Namespace)}
```
Comment (bowei) [inline cmd/e2e-test/asm_test.go:306]: could use "..." + sSkip.Namespace instead of fmt [minor]

File: cmd/e2e-test/asm_test.go [hunk 5]
```diff
@@ -285,3 +287,95 @@ func TestNoIstioASM(t *testing.T) {
...
+			if err := e2e.EnsureConfigMap(s, asmConfigNamespace, asmConfigName,
+				asmConfig); err != nil {
+				t.Error(err)
```
Comment (bowei) [inline cmd/e2e-test/asm_test.go:309]: t.Error or t.Fatal?

File: cmd/e2e-test/asm_test.go (hunk 1 above)
Comment (swetharepakula) [reply cmd/e2e-test/asm_test.go:293]: I am not sure, this is a duplication of the original ASM test with removal of the destination rule pieces. I will test locally and ensure we don't need that.

File: cmd/e2e-test/asm_test.go (hunk 2 above)
Comment (swetharepakula) [reply cmd/e2e-test/asm_test.go]: Extended the sandbox to be able to create and track additional namespaces.

**Pull request closed without merging**